doc/develop/tools/clion.rst

```diff
@@ -0,0 +1,267 @@
+.. _clion:
+
+..
+   Copyright 2023 Marina Kalashina <marina.kalashina@jetbrains.com>
+
+
+CLion
+#####
+
+`CLion <https://www.jetbrains.com/clion/>`_ is a cross-platform C/C++ IDE that
+supports multi-threaded RTOS debugging.
+
+This guide describes the process of setting up, building, and debugging
+Zephyr's example-application_ in CLion.
+
+.. _example-application: https://github.com/zephyrproject-rtos/example-application
+
+
+Get CLion
+*********
+
+Download_ and install CLion.
+
+.. _Download: https://www.jetbrains.com/clion/download
+
+Initialize a new workspace
+**************************
+
+#. Before you start, make sure to prepare a Zephyr development environment:
+
+   * `Install dependencies`_
+
+      .. _Install dependencies: https://docs.zephyrproject.org/latest/develop/getting_started/index.html#install-dependencies
+
+   * `Get Zephyr and install Python dependencies`_
+
+      .. _Get Zephyr and install Python dependencies: https://docs.zephyrproject.org/latest/develop/getting_started/index.html#get-zephyr-and-install-python-dependencies
+
+   * `Install Zephyr SDK`_
+
+      .. _Install Zephyr SDK: https://docs.zephyrproject.org/latest/develop/getting_started/index.html#install-zephyr-sdk
+
+#. Run the following command to initialize the workspace folder
+   (``my-workspace``) and clone the example application into it:
+
+   .. code-block::
+
+      west init -m https://github.com/zephyrproject-rtos/example-application
+      --mr main my-workspace
+
+#. Update Zephyr modules:
+
+   .. code-block::
+
+      cd my-workspace
+      west update
+
+
+Open the project in CLion
+**************************
+
+#. In CLion, click **Open** on the Welcome screen or select **File | Open**
+   from the main menu.
+
+#. Locate the workspace folder (``my-workspace``) and select it.
```

Comment:
I recall we tried opening a project instead of workspace, but that didn't work for us, unfortunately. So since we couldn't test that properly, we decided not to include that in the instructions and only write about what actually worked.